[elasticsearch] Handle Bad Requests caused by syntax errors in search queries

<!--
- Please follow the issue template below for bug reports.
- If you have a support request rather than a bug, please use [Stack Overflow](http://stackoverflow.com/questions/tagged/jhipster) with the JHipster tag.
- For bug reports it is mandatory to run the command `jhipster info` in your project's root folder, and paste the result here.
- Tickets opened without any of these pieces of information will be **closed** without any explanation.
-
- If this is a potential security issue, please read our security policy and contact us privately first: https://github.com/jhipster/generator-jhipster/security/policy
-->

##### **Overview of the issue**

<!-- Explain the bug, if an error is being thrown a stack trace helps -->
If a user submits a query with invalid syntax from the main CRUD screen, a message reporting an "Internal server error." pops up and the loading icon of the "Refresh list" button spins forever. In this state no further search is possible. The user has to reload the main CRUD screen to continue searching. On server side a stacktrace is logged.

  <details>
  <summary>Stacktrace</summary>
<pre>
2023-04-04T10:18:41.222+02:00 ERROR 21804 --- [  XNIO-1 task-5] o.z.problem.spring.common.AdviceTraits   : Internal Server Error

org.springframework.data.elasticsearch.RestStatusException: Elasticsearch exception [type=search_phase_execution_exception, reason=all shards failed]; nested exception is ElasticsearchStatusException[Elasticsearch exception [type=search_phase_execution_exception, reason=all shards failed]]
        at org.springframework.data.elasticsearch.core.ElasticsearchExceptionTranslator.translateExceptionIfPossible(ElasticsearchExceptionTranslator.java:69)
        at org.springframework.data.elasticsearch.core.ElasticsearchRestTemplate.translateException(ElasticsearchRestTemplate.java:601)
        at org.springframework.data.elasticsearch.core.ElasticsearchRestTemplate.execute(ElasticsearchRestTemplate.java:584)
        at org.springframework.data.elasticsearch.core.ElasticsearchRestTemplate.search(ElasticsearchRestTemplate.java:399)
        at org.springframework.data.elasticsearch.core.AbstractElasticsearchTemplate.search(AbstractElasticsearchTemplate.java:418)
        at com.mycompany.myapp.repository.search.TestEntitySearchRepositoryInternalImpl.search(TestEntitySearchRepository.java:49)
        at com.mycompany.myapp.repository.search.TestEntitySearchRepositoryInternalImpl.search(TestEntitySearchRepository.java:44)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.base/java.lang.reflect.Method.invoke(Method.java:566)
        at org.springframework.data.repository.core.support.RepositoryMethodInvoker$RepositoryFragmentMethodInvoker.lambda$new$0(RepositoryMethodInvoker.java:289)
        at org.springframework.data.repository.core.support.RepositoryMethodInvoker.doInvoke(RepositoryMethodInvoker.java:137)
        at org.springframework.data.repository.core.support.RepositoryMethodInvoker.invoke(RepositoryMethodInvoker.java:121)
        at org.springframework.data.repository.core.support.RepositoryComposition$RepositoryFragments.invoke(RepositoryComposition.java:530)
        at org.springframework.data.repository.core.support.RepositoryComposition.invoke(RepositoryComposition.java:286)        at org.springframework.data.repository.core.support.RepositoryFactorySupport$ImplementationMethodExecutionInterceptor.invoke(RepositoryFactorySupport.java:640)
        at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:186)
        at org.springframework.data.repository.core.support.QueryExecutorMethodInterceptor.doInvoke(QueryExecutorMethodInterceptor.java:164)
        at org.springframework.data.repository.core.support.QueryExecutorMethodInterceptor.invoke(QueryExecutorMethodInterceptor.java:139)
        at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:186)
        at org.springframework.aop.interceptor.ExposeInvocationInterceptor.invoke(ExposeInvocationInterceptor.java:97)
        at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:186)
        at org.springframework.aop.framework.JdkDynamicAopProxy.invoke(JdkDynamicAopProxy.java:215)
        at com.sun.proxy.$Proxy214.search(Unknown Source)
        at com.mycompany.myapp.web.rest.TestEntityResource.searchTestEntities(TestEntityResource.java:206)
        at com.mycompany.myapp.web.rest.TestEntityResource$$FastClassBySpringCGLIB$$96114660.invoke(<generated>)
        at org.springframework.cglib.proxy.MethodProxy.invoke(MethodProxy.java:218)
        at org.springframework.aop.framework.CglibAopProxy$CglibMethodInvocation.invokeJoinpoint(CglibAopProxy.java:793)        at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:163)
        at org.springframework.aop.framework.CglibAopProxy$CglibMethodInvocation.proceed(CglibAopProxy.java:763)
        at org.springframework.aop.aspectj.AspectJAfterThrowingAdvice.invoke(AspectJAfterThrowingAdvice.java:64)
        at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:186)
        at org.springframework.aop.framework.CglibAopProxy$CglibMethodInvocation.proceed(CglibAopProxy.java:763)
        at org.springframework.aop.aspectj.MethodInvocationProceedingJoinPoint.proceed(MethodInvocationProceedingJoinPoint.java:89)
        at com.mycompany.myapp.aop.logging.LoggingAspect.logAround(LoggingAspect.java:105)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.base/java.lang.reflect.Method.invoke(Method.java:566)
        at org.springframework.aop.aspectj.AbstractAspectJAdvice.invokeAdviceMethodWithGivenArgs(AbstractAspectJAdvice.java:634)
        at org.springframework.aop.aspectj.AbstractAspectJAdvice.invokeAdviceMethod(AbstractAspectJAdvice.java:624)
        at org.springframework.aop.aspectj.AspectJAroundAdvice.invoke(AspectJAroundAdvice.java:72)
        at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:186)
        at org.springframework.aop.framework.CglibAopProxy$CglibMethodInvocation.proceed(CglibAopProxy.java:763)
        at org.springframework.transaction.interceptor.TransactionInterceptor$1.proceedWithInvocation(TransactionInterceptor.java:123)
        at org.springframework.transaction.interceptor.TransactionAspectSupport.invokeWithinTransaction(TransactionAspectSupport.java:388)
        at org.springframework.transaction.interceptor.TransactionInterceptor.invoke(TransactionInterceptor.java:119)
        at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:186)
        at org.springframework.aop.framework.CglibAopProxy$CglibMethodInvocation.proceed(CglibAopProxy.java:763)
        at org.springframework.aop.interceptor.ExposeInvocationInterceptor.invoke(ExposeInvocationInterceptor.java:97)
        at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:186)
        at org.springframework.aop.framework.CglibAopProxy$CglibMethodInvocation.proceed(CglibAopProxy.java:763)
        at org.springframework.aop.framework.CglibAopProxy$DynamicAdvisedInterceptor.intercept(CglibAopProxy.java:708)
        at com.mycompany.myapp.web.rest.TestEntityResource$$EnhancerBySpringCGLIB$$76cc6a1c.searchTestEntities(<generated>)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.base/java.lang.reflect.Method.invoke(Method.java:566)
        at org.springframework.web.method.support.InvocableHandlerMethod.doInvoke(InvocableHandlerMethod.java:205)
        at org.springframework.web.method.support.InvocableHandlerMethod.invokeForRequest(InvocableHandlerMethod.java:150)
        at org.springframework.web.servlet.mvc.method.annotation.ServletInvocableHandlerMethod.invokeAndHandle(ServletInvocableHandlerMethod.java:117)
        at org.springframework.web.servlet.mvc.method.annotation.RequestMappingHandlerAdapter.invokeHandlerMethod(RequestMappingHandlerAdapter.java:895)
        at org.springframework.web.servlet.mvc.method.annotation.RequestMappingHandlerAdapter.handleInternal(RequestMappingHandlerAdapter.java:808)
        at org.springframework.web.servlet.mvc.method.AbstractHandlerMethodAdapter.handle(AbstractHandlerMethodAdapter.java:87)
        at org.springframework.web.servlet.DispatcherServlet.doDispatch(DispatcherServlet.java:1070)
        at org.springframework.web.servlet.DispatcherServlet.doService(DispatcherServlet.java:963)
        at org.springframework.web.servlet.FrameworkServlet.processRequest(FrameworkServlet.java:1006)
        at org.springframework.web.servlet.FrameworkServlet.doGet(FrameworkServlet.java:898)
        at javax.servlet.http.HttpServlet.service(HttpServlet.java:497)
        at org.springframework.web.servlet.FrameworkServlet.service(FrameworkServlet.java:883)
        at javax.servlet.http.HttpServlet.service(HttpServlet.java:584)
        at io.undertow.servlet.handlers.ServletHandler.handleRequest(ServletHandler.java:74)
        at io.undertow.servlet.handlers.FilterHandler$FilterChainImpl.doFilter(FilterHandler.java:129)
        at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:111)
        at io.undertow.servlet.core.ManagedFilter.doFilter(ManagedFilter.java:61)
        at io.undertow.servlet.handlers.FilterHandler$FilterChainImpl.doFilter(FilterHandler.java:131)
        at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:337)
        at org.springframework.security.web.access.intercept.FilterSecurityInterceptor.invoke(FilterSecurityInterceptor.java:115)
        at org.springframework.security.web.access.intercept.FilterSecurityInterceptor.doFilter(FilterSecurityInterceptor.java:81)
        at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:346)
        at org.springframework.security.web.access.ExceptionTranslationFilter.doFilter(ExceptionTranslationFilter.java:122)
        at org.springframework.security.web.access.ExceptionTranslationFilter.doFilter(ExceptionTranslationFilter.java:116)
        at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:346)
        at org.springframework.security.web.session.SessionManagementFilter.doFilter(SessionManagementFilter.java:126)
        at org.springframework.security.web.session.SessionManagementFilter.doFilter(SessionManagementFilter.java:81)
        at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:346)
        at org.springframework.security.web.authentication.AnonymousAuthenticationFilter.doFilter(AnonymousAuthenticationFilter.java:109)
        at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:346)
        at org.springframework.security.web.servletapi.SecurityContextHolderAwareRequestFilter.doFilter(SecurityContextHolderAwareRequestFilter.java:149)
        at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:346)
        at org.springframework.security.web.savedrequest.RequestCacheAwareFilter.doFilter(RequestCacheAwareFilter.java:63)
        at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:346)
        at org.springframework.security.web.authentication.www.BasicAuthenticationFilter.doFilterInternal(BasicAuthenticationFilter.java:166)
        at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:117)
        at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:346)
        at com.mycompany.myapp.security.jwt.JWTFilter.doFilter(JWTFilter.java:37)
        at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:346)
        at org.springframework.web.filter.CorsFilter.doFilterInternal(CorsFilter.java:91)
        at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:117)
        at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:346)
        at org.springframework.security.web.authentication.logout.LogoutFilter.doFilter(LogoutFilter.java:103)
        at org.springframework.security.web.authentication.logout.LogoutFilter.doFilter(LogoutFilter.java:89)
        at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:346)
        at org.springframework.security.web.header.HeaderWriterFilter.doHeadersAfter(HeaderWriterFilter.java:90)
        at org.springframework.security.web.header.HeaderWriterFilter.doFilterInternal(HeaderWriterFilter.java:75)
        at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:117)
        at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:346)
        at org.springframework.security.web.context.SecurityContextPersistenceFilter.doFilter(SecurityContextPersistenceFilter.java:112)
        at org.springframework.security.web.context.SecurityContextPersistenceFilter.doFilter(SecurityContextPersistenceFilter.java:82)
        at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:346)
        at org.springframework.security.web.context.request.async.WebAsyncManagerIntegrationFilter.doFilterInternal(WebAsyncManagerIntegrationFilter.java:55)
        at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:117)
        at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:346)
        at org.springframework.security.web.session.DisableEncodeUrlFilter.doFilterInternal(DisableEncodeUrlFilter.java:42)
        at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:117)
        at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:346)
        at org.springframework.security.web.FilterChainProxy.doFilterInternal(FilterChainProxy.java:221)
        at org.springframework.security.web.FilterChainProxy.doFilter(FilterChainProxy.java:186)
        at org.springframework.web.filter.DelegatingFilterProxy.invokeDelegate(DelegatingFilterProxy.java:354)
        at org.springframework.web.filter.DelegatingFilterProxy.doFilter(DelegatingFilterProxy.java:267)
        at io.undertow.servlet.core.ManagedFilter.doFilter(ManagedFilter.java:61)
        at io.undertow.servlet.handlers.FilterHandler$FilterChainImpl.doFilter(FilterHandler.java:131)
        at org.springframework.web.filter.RequestContextFilter.doFilterInternal(RequestContextFilter.java:100)
        at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:117)
        at io.undertow.servlet.core.ManagedFilter.doFilter(ManagedFilter.java:61)
        at io.undertow.servlet.handlers.FilterHandler$FilterChainImpl.doFilter(FilterHandler.java:131)
        at org.springframework.web.filter.FormContentFilter.doFilterInternal(FormContentFilter.java:93)
        at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:117)
        at io.undertow.servlet.core.ManagedFilter.doFilter(ManagedFilter.java:61)
        at io.undertow.servlet.handlers.FilterHandler$FilterChainImpl.doFilter(FilterHandler.java:131)
        at org.springframework.boot.actuate.metrics.web.servlet.WebMvcMetricsFilter.doFilterInternal(WebMvcMetricsFilter.java:96)
        at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:117)
        at io.undertow.servlet.core.ManagedFilter.doFilter(ManagedFilter.java:61)
        at io.undertow.servlet.handlers.FilterHandler$FilterChainImpl.doFilter(FilterHandler.java:131)
        at org.springframework.web.filter.CharacterEncodingFilter.doFilterInternal(CharacterEncodingFilter.java:201)
        at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:117)
        at io.undertow.servlet.core.ManagedFilter.doFilter(ManagedFilter.java:61)
        at io.undertow.servlet.handlers.FilterHandler$FilterChainImpl.doFilter(FilterHandler.java:131)
        at io.undertow.servlet.handlers.FilterHandler.handleRequest(FilterHandler.java:84)
        at io.undertow.servlet.handlers.security.ServletSecurityRoleHandler.handleRequest(ServletSecurityRoleHandler.java:62)
        at io.undertow.servlet.handlers.ServletChain$1.handleRequest(ServletChain.java:68)
        at io.undertow.servlet.handlers.ServletDispatchingHandler.handleRequest(ServletDispatchingHandler.java:36)
        at io.undertow.servlet.handlers.RedirectDirHandler.handleRequest(RedirectDirHandler.java:68)
        at io.undertow.servlet.handlers.security.SSLInformationAssociationHandler.handleRequest(SSLInformationAssociationHandler.java:117)
        at io.undertow.servlet.handlers.security.ServletAuthenticationCallHandler.handleRequest(ServletAuthenticationCallHandler.java:57)
        at io.undertow.server.handlers.PredicateHandler.handleRequest(PredicateHandler.java:43)
        at io.undertow.security.handlers.AbstractConfidentialityHandler.handleRequest(AbstractConfidentialityHandler.java:46)
        at io.undertow.servlet.handlers.security.ServletConfidentialityConstraintHandler.handleRequest(ServletConfidentialityConstraintHandler.java:64)
        at io.undertow.security.handlers.AuthenticationMechanismsHandler.handleRequest(AuthenticationMechanismsHandler.java:60)
        at io.undertow.servlet.handlers.security.CachedAuthenticatedSessionHandler.handleRequest(CachedAuthenticatedSessionHandler.java:77)
        at io.undertow.security.handlers.AbstractSecurityContextAssociationHandler.handleRequest(AbstractSecurityContextAssociationHandler.java:43)
        at io.undertow.server.handlers.PredicateHandler.handleRequest(PredicateHandler.java:43)
        at io.undertow.servlet.handlers.SendErrorPageHandler.handleRequest(SendErrorPageHandler.java:52)
        at io.undertow.server.handlers.PredicateHandler.handleRequest(PredicateHandler.java:43)
        at io.undertow.servlet.handlers.SessionRestoringHandler.handleRequest(SessionRestoringHandler.java:119)
        at io.undertow.servlet.handlers.ServletInitialHandler.handleFirstRequest(ServletInitialHandler.java:275)
        at io.undertow.servlet.handlers.ServletInitialHandler.access$100(ServletInitialHandler.java:79)
        at io.undertow.servlet.handlers.ServletInitialHandler$2.call(ServletInitialHandler.java:134)
        at io.undertow.servlet.handlers.ServletInitialHandler$2.call(ServletInitialHandler.java:131)
        at io.undertow.servlet.core.ServletRequestContextThreadSetupAction$1.call(ServletRequestContextThreadSetupAction.java:48)
        at io.undertow.servlet.core.ContextClassLoaderSetupAction$1.call(ContextClassLoaderSetupAction.java:43)
        at io.undertow.servlet.handlers.ServletInitialHandler.dispatchRequest(ServletInitialHandler.java:255)
        at io.undertow.servlet.handlers.ServletInitialHandler.access$000(ServletInitialHandler.java:79)
        at io.undertow.servlet.handlers.ServletInitialHandler$1.handleRequest(ServletInitialHandler.java:100)
        at io.undertow.server.Connectors.executeRootHandler(Connectors.java:387)
        at io.undertow.server.HttpServerExchange$1.run(HttpServerExchange.java:852)
        at org.jboss.threads.ContextClassLoaderSavingRunnable.run(ContextClassLoaderSavingRunnable.java:35)
        at org.jboss.threads.EnhancedQueueExecutor.safeRun(EnhancedQueueExecutor.java:2019)
        at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.doRunTask(EnhancedQueueExecutor.java:1558)
        at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1449)
        at org.xnio.XnioWorker$WorkerThreadFactory$1$1.run(XnioWorker.java:1282)
        at java.base/java.lang.Thread.run(Thread.java:829)
Caused by: org.elasticsearch.ElasticsearchStatusException: Elasticsearch exception [type=search_phase_execution_exception, reason=all shards failed]
        at org.elasticsearch.rest.BytesRestResponse.errorFromXContent(BytesRestResponse.java:178)
        at org.elasticsearch.client.RestHighLevelClient.parseEntity(RestHighLevelClient.java:2484)
        at org.elasticsearch.client.RestHighLevelClient.parseResponseException(RestHighLevelClient.java:2461)
        at org.elasticsearch.client.RestHighLevelClient.internalPerformRequest(RestHighLevelClient.java:2184)
        at org.elasticsearch.client.RestHighLevelClient.performRequest(RestHighLevelClient.java:2137)
        at org.elasticsearch.client.RestHighLevelClient.performRequestAndParseEntity(RestHighLevelClient.java:2105)
        at org.elasticsearch.client.RestHighLevelClient.search(RestHighLevelClient.java:1367)
        at org.springframework.data.elasticsearch.core.ElasticsearchRestTemplate.lambda$search$14(ElasticsearchRestTemplate.java:399)
        at org.springframework.data.elasticsearch.core.ElasticsearchRestTemplate.execute(ElasticsearchRestTemplate.java:582)
        ... 170 common frames omitted
        Suppressed: org.elasticsearch.client.ResponseException: method [POST], host [http://localhost:9200], URI [/testentity/_search?typed_keys=true&max_concurrent_shard_requests=5&search_type=query_then_fetch&batched_reduce_size=512], status line [HTTP/1.1 400 Bad Request]
Warnings: [Elasticsearch built-in security features are not enabled. Without authentication, your cluster could be accessible to anyone. See https://www.elastic.co/guide/en/elasticsearch/reference/7.17/security-minimal-setup.html to enable security.]
{"error":{"root_cause":[{"type":"query_shard_exception","reason":"Failed to parse query [08/15]","index_uuid":"3WK3pjRIQ6mpnrgVfDuKWA","index":"testentity"}],"type":"search_phase_execution_exception","reason":"all shards failed","phase":"query","grouped":true,"failed_shards":[{"shard":0,"index":"testentity","node":"JAsMkNZ-Sb6gsWN5BdXKrg","reason":{"type":"query_shard_exception","reason":"Failed to parse query [08/15]","index_uuid":"3WK3pjRIQ6mpnrgVfDuKWA","index":"testentity","caused_by":{"type":"parse_exception","reason":"Cannot parse '08/15': Lexical error at line 1, column 6.  Encountered: <EOF> after : \"/15\"","caused_by":{"type":"token_mgr_error","reason":"Lexical error at line 1, column 6.  Encountered: <EOF> after : \"/15\""}}}}]},"status":400}
                at org.elasticsearch.client.RestClient.convertResponse(RestClient.java:346)
                at org.elasticsearch.client.RestClient.performRequest(RestClient.java:312)
                at org.elasticsearch.client.RestClient.performRequest(RestClient.java:287)
                at org.elasticsearch.client.RestHighLevelClient.performClientRequest(RestHighLevelClient.java:2699)
                at org.elasticsearch.client.RestHighLevelClient.internalPerformRequest(RestHighLevelClient.java:2171)
                ... 175 common frames omitted
</pre>
  </details>

##### **Motivation for or Use Case**

<!-- Explain why this is a bug for you -->
It should not be possible for a user to generate an internal server error by submitting a simple value.

##### **Reproduce the error**

<!-- For bug reports, an unambiguous set of steps to reproduce the error -->
Use JHipster 7.9.3, use same elasticsearch version as defined in the docker config (elasticsearch.yml) of JHipster 7.9.3
Create an application with 
- angular
- any database
- at least one entity defined
- elasticsearch enabled

The error can be reproduced even with no data in the database or elasticsearch index.

Submit a search with an invalid lucene syntax e.g.   08/15   (without quotes)

##### **Related issues**

<!-- Has a similar issue been reported before? Please search both closed & open issues -->

##### **Suggest a Fix**

<!-- If you can't fix the bug yourself, perhaps you can point to what might be
  causing the problem (line of code or commit) -->

Elasticsearch exceptions with reason "Failed to parse query" should be filtered and should not be logged as errors. A dedicated error message should be reported to the user and the error should be handled properly.

Each ui framework may be affected. I tested only with angular.

It would be nice to have the supported query syntax of JHipster (https://lucene.apache.org/core/8_11_2/queryparser/org/apache/lucene/queryparser/classic/package-summary.html#package.description) linked at the elasticsearch documentation https://www.jhipster.tech/using-elasticsearch/ .

##### **JHipster Version(s)**

7.9.3
<!--
Which version of JHipster are you using, is it a regression?
-->

##### **JHipster configuration**

<!--
To provide all information we need, you should run `jhipster info` in the project root folder, and
copy/paste the result here.
The `.yo-rc.json` file generated in the root folder is mandatory for bug reports. This will help us to replicate the scenario.
You should remove any sensitive information like the rememberMe key or the jwtSecretKey key.
---
If you have a JDL please wrap it in below structure
  <details>
  <summary>JDL definitions</summary>
  <pre>
     JDL content here
  </pre>
  </details>
-->
  <details>
  <summary>JDL definitions</summary>
  <pre>
application {
  config {
    applicationType monolith
    authenticationType jwt
    baseName myapp
    blueprints []
    buildTool maven
    cacheProvider ehcache
    clientFramework angularX
    clientPackageManager npm
    clientTheme none
    creationTimestamp 1680595130781
    databaseType sql
    devDatabaseType h2Disk
    dtoSuffix DTO
    enableHibernateCache true
    enableSwaggerCodegen false
    enableTranslation true
    jhiPrefix jhi
    jhipsterVersion "7.9.3"
    languages [en]
    messageBroker false
    nativeLanguage en
    otherModules []
    packageName com.mycompany.myapp
    prodDatabaseType postgresql
    reactive false
    searchEngine elasticsearch
    serverPort 8080
    serviceDiscoveryType no
    skipClient false
    skipUserManagement false
    testFrameworks []
    websocket false
  }
  entities TestEntity
}
entity TestEntity {
  text String
}
  </pre>
  </details>

##### **Entity configuration(s) `entityName.json` files generated in the `.jhipster` directory**

<!--
If the error is during an entity creation or associated with a specific entity.
If you are using JDL, please share that configuration as well.
-->

##### **Browsers and Operating System**

not relevant for this issue
<!-- What OS are you on? is this a problem with all browsers or only IE8? -->

- [x] Checking this box is mandatory (this is just to show you read everything)

<!-- Love JHipster? Please consider supporting our collective:
👉  https://opencollective.com/generator-jhipster/donate -->

--- a/generators/angular/templates/src/main/webapp/app/entities/_entityFolder/service/_entityFile.service.spec.ts.ejs
+++ b/generators/angular/templates/src/main/webapp/app/entities/_entityFolder/service/_entityFile.service.spec.ts.ejs
@@ -134,6 +134,22 @@ describe('<%= entityAngularName %> Service', () => {
         });
 <%_ } _%>
 
+<%_ if (searchEngineElasticsearch) { _%>
+        it('should handle exceptions for searching a <%= entityAngularName %>', () => {
+            const queryObject: any = {
+                page: 0,
+                size: 20,
+                query: '',
+                sort: [],
+            };
+            service.search(queryObject).subscribe(() => expectedResult);
+
+            const req = httpMock.expectOne({ method: 'GET' });
+            req.flush(null, { status: 500, statusText: 'Internal Server Error' });
+            expect(expectedResult).toBe(null);
+        });
+<%_ } _%>
+
         describe('add<%= entityAngularName %>ToCollectionIfMissing', () => {
             it('should add a <%= entityAngularName %> to an empty array', () => {
                 const <%= entityInstance %>: I<%= entityAngularName %> = sampleWithRequiredData;

--- a/generators/angular/templates/src/main/webapp/app/entities/_entityFolder/service/_entityFile.service.ts.ejs
+++ b/generators/angular/templates/src/main/webapp/app/entities/_entityFolder/service/_entityFile.service.ts.ejs
@@ -24,9 +24,19 @@ if (paginationPagination || paginationInfiniteScroll) {
 _%>
 import { Injectable } from '@angular/core';
 import { HttpClient, HttpResponse } from '@angular/common/http';
-import { Observable } from 'rxjs';
-<%_ if (anyFieldIsDateDerived) { _%>
+import { Observable<%_ if (searchEngineElasticsearch) { _%>, asapScheduler, scheduled<%_ } _%> } from 'rxjs';
+
+<%_ if (searchEngineElasticsearch) {
+  if (anyFieldIsDateDerived) { _%>
+import { catchError, map } from 'rxjs/operators';
+<% } else { %>
+import { catchError } from 'rxjs/operators';
+<%_ }
+   } else if (anyFieldIsDateDerived) { _%>
 import { map } from 'rxjs/operators';
+<%_ } _%>
+
+<%_ if (anyFieldIsDateDerived) { _%>
 import dayjs from 'dayjs/esm';
 <%_ } _%>
 
@@ -125,7 +135,7 @@ export class <%= entityAngularName %>Service {
     search(req: <%= searchType %>): Observable<EntityArrayResponseType> {
         const options = createRequestOption(req);
         return this.http.get<<%= entityRestName %>[]>(this.resourceSearchUrl, { params: options, observe: 'response' })
-            <% if (anyFieldIsDateDerived) { %>.pipe(map(res => this.convertResponseArrayFromServer(res)))<% } %>;
+            .pipe(<% if (anyFieldIsDateDerived) { %>map(res => this.convertResponseArrayFromServer(res)), <% } %>catchError(() => scheduled([new HttpResponse<I<%= entityAngularName %>[]>()], asapScheduler)));
     }
 <%_ } _%>
 

--- a/generators/elasticsearch/files.mjs
+++ b/generators/elasticsearch/files.mjs
@@ -24,7 +24,11 @@ export const files = {
     {
       path: `${SERVER_MAIN_SRC_DIR}package/`,
       renameTo: moveToJavaPackageSrcDir,
-      templates: ['config/ElasticsearchConfiguration.java'],
+      templates: [
+        'config/ElasticsearchConfiguration.java',
+        'web/rest/errors/ElasticsearchExceptionMapper.java',
+        'web/rest/errors/QuerySyntaxException.java',
+      ],
     },
     {
       condition: generator => generator.generateBuiltInUserEntity || generator.authenticationTypeOauth2,
@@ -39,6 +43,7 @@ export const files = {
         'config/ElasticsearchTestConfiguration.java',
         'config/ElasticsearchTestContainer.java',
         'config/EmbeddedElasticsearch.java',
+        'web/rest/errors/ElasticsearchExceptionMapperTest.java',
       ],
     },
   ],

--- a/generators/elasticsearch/templates/src/main/java/package/web/rest/errors/ElasticsearchExceptionMapper.java.ejs
+++ b/generators/elasticsearch/templates/src/main/java/package/web/rest/errors/ElasticsearchExceptionMapper.java.ejs
@@ -1,0 +1,49 @@
+<%#
+ Copyright 2013-2023 the original author or authors from the JHipster project.
+
+ This file is part of the JHipster project, see https://www.jhipster.tech/
+ for more information.
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+      https://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+-%>
+package <%= packageName %>.web.rest.errors;
+
+import co.elastic.clients.elasticsearch._types.ElasticsearchException;
+import co.elastic.clients.elasticsearch._types.ErrorCause;
+import org.springframework.data.elasticsearch.UncategorizedElasticsearchException;
+
+import java.util.List;
+
+public class ElasticsearchExceptionMapper {
+
+    private ElasticsearchExceptionMapper() {}
+
+    public static RuntimeException mapException(RuntimeException originalException) {
+        RuntimeException e = originalException;
+        if (e.getCause() instanceof UncategorizedElasticsearchException) {
+            e = (UncategorizedElasticsearchException) e.getCause();
+        }
+        if (e.getCause() instanceof ElasticsearchException) {
+            ElasticsearchException esException = (ElasticsearchException) e.getCause();
+            List<ErrorCause> rootCause = esException.response().error().rootCause();
+            if (!rootCause.isEmpty()) {
+                String reason = rootCause.get(0).reason();
+                if (reason != null && reason.startsWith("Failed to parse query [")) {
+                    return new QuerySyntaxException();
+                }
+            }
+        }
+
+        return originalException;
+    }
+}

--- a/generators/elasticsearch/templates/src/main/java/package/web/rest/errors/QuerySyntaxException.java.ejs
+++ b/generators/elasticsearch/templates/src/main/java/package/web/rest/errors/QuerySyntaxException.java.ejs
@@ -1,0 +1,30 @@
+<%#
+ Copyright 2013-2023 the original author or authors from the JHipster project.
+
+ This file is part of the JHipster project, see https://www.jhipster.tech/
+ for more information.
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+      https://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+-%>
+package <%= packageName %>.web.rest.errors;
+
+@SuppressWarnings("java:S110") // Inheritance tree of classes should not be too deep
+public class QuerySyntaxException extends BadRequestAlertException {
+
+    private static final long serialVersionUID = 1L;
+
+    public QuerySyntaxException() {
+        super("Invalid query syntax!", "elasticseach", "querySyntaxError");
+    }
+
+}

--- a/generators/elasticsearch/templates/src/main/java/package/web/web/rest/errors/ElasticsearchExceptionMapper.java.ejs
+++ b/generators/elasticsearch/templates/src/main/java/package/web/web/rest/errors/ElasticsearchExceptionMapper.java.ejs
@@ -1,0 +1,49 @@
+<%#
+ Copyright 2013-2023 the original author or authors from the JHipster project.
+
+ This file is part of the JHipster project, see https://www.jhipster.tech/
+ for more information.
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+      https://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+-%>
+package <%= packageName %>.web.rest.errors;
+
+import co.elastic.clients.elasticsearch._types.ElasticsearchException;
+import co.elastic.clients.elasticsearch._types.ErrorCause;
+import org.springframework.data.elasticsearch.UncategorizedElasticsearchException;
+
+import java.util.List;
+
+public class ElasticsearchExceptionMapper {
+
+    private ElasticsearchExceptionMapper() {}
+
+    public static RuntimeException mapException(RuntimeException originalException) {
+        RuntimeException e = originalException;
+        if (e.getCause() instanceof UncategorizedElasticsearchException) {
+            e = (UncategorizedElasticsearchException) e.getCause();
+        }
+        if (e.getCause() instanceof ElasticsearchException) {
+            ElasticsearchException esException = (ElasticsearchException) e.getCause();
+            List<ErrorCause> rootCause = esException.response().error().rootCause();
+            if (!rootCause.isEmpty()) {
+                String reason = rootCause.get(0).reason();
+                if (reason != null && reason.startsWith("Failed to parse query [")) {
+                    return new QuerySyntaxException();
+                }
+            }
+        }
+
+        return originalException;
+    }
+}

--- a/generators/elasticsearch/templates/src/main/java/package/web/web/rest/errors/QuerySyntaxException.java.ejs
+++ b/generators/elasticsearch/templates/src/main/java/package/web/web/rest/errors/QuerySyntaxException.java.ejs
@@ -1,0 +1,30 @@
+<%#
+ Copyright 2013-2023 the original author or authors from the JHipster project.
+
+ This file is part of the JHipster project, see https://www.jhipster.tech/
+ for more information.
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+      https://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+-%>
+package <%= packageName %>.web.rest.errors;
+
+@SuppressWarnings("java:S110") // Inheritance tree of classes should not be too deep
+public class QuerySyntaxException extends BadRequestAlertException {
+
+    private static final long serialVersionUID = 1L;
+
+    public QuerySyntaxException() {
+        super("Invalid query syntax!", "elasticseach", "querySyntaxError");
+    }
+
+}

--- a/generators/elasticsearch/templates/src/test/java/package/web/rest/errors/ElasticsearchExceptionMapperTest.java.ejs
+++ b/generators/elasticsearch/templates/src/test/java/package/web/rest/errors/ElasticsearchExceptionMapperTest.java.ejs
@@ -1,0 +1,45 @@
+<%#
+ Copyright 2013-2023 the original author or authors from the JHipster project.
+
+ This file is part of the JHipster project, see https://www.jhipster.tech/
+ for more information.
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+      https://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+-%>
+package <%= packageName %>.web.rest.errors;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+import co.elastic.clients.elasticsearch._types.ElasticsearchException;
+import co.elastic.clients.elasticsearch._types.ErrorCause;
+import co.elastic.clients.elasticsearch._types.ErrorResponse;
+import org.junit.jupiter.api.Test;
+import org.springframework.data.elasticsearch.UncategorizedElasticsearchException;
+
+class ElasticsearchExceptionMapperTest {
+
+    @Test
+    void testMapException() {
+        ErrorCause rootCause = new ErrorCause.Builder().reason("Failed to parse query [").build();
+        ErrorResponse response = new ErrorResponse.Builder()
+            .error(new ErrorCause.Builder().reason("").rootCause(rootCause).build())
+            .status(400)
+            .build();
+        ElasticsearchException esException = new ElasticsearchException("", response);
+        UncategorizedElasticsearchException cause = new UncategorizedElasticsearchException("", esException);
+        assertInstanceOf(
+            QuerySyntaxException.class,
+            ElasticsearchExceptionMapper.mapException(new UncategorizedElasticsearchException("", cause))
+        );
+    }
+}

--- a/generators/languages/templates/src/main/webapp/i18n/de/global.json.ejs
+++ b/generators/languages/templates/src/main/webapp/i18n/de/global.json.ejs
@@ -158,6 +158,9 @@
         "idnull": "Ungültige ID",
         "idinvalid": "Ungültige ID",
         "idnotfound": "ID konnte nicht gefunden werden",
+<%_ if (searchEngineElasticsearch) { _%>
+        "querySyntaxError": "Ungültige Suchanfrage",
+<%_ } _%>
         "file": {
             "could.not.extract": "Datei konnte nicht extrahiert werden",
             "not.image": "Es wird erwartet, dass die Datei ein Bild ist. Die Datei ist jedoch vom Typ \"{{ fileType }}\""

--- a/generators/languages/templates/src/main/webapp/i18n/en/global.json.ejs
+++ b/generators/languages/templates/src/main/webapp/i18n/en/global.json.ejs
@@ -159,6 +159,9 @@
         "idnull": "Invalid ID",
         "idinvalid": "Invalid ID",
         "idnotfound": "ID cannot be found",
+<%_ if (searchEngineElasticsearch) { _%>
+        "querySyntaxError": "Invalid search query",
+<%_ } _%>
         "file": {
             "could.not.extract": "Could not extract file",
             "not.image": "File was expected to be an image but was found to be \"{{ fileType }}\""

--- a/generators/server/templates/entity/src/main/java/package/common/search_stream_template.ejs
+++ b/generators/server/templates/entity/src/main/java/package/common/search_stream_template.ejs
@@ -19,24 +19,32 @@
 <%_
 const mapper = entityInstance  + 'Mapper';
 const entityToDtoReference = mapper + '::' + 'toDto'; %>
+        try {
 <%_ if (!viaService) { _%>
   <%_ if (reactive) { _%>
-        return <%= entityInstance %><% if (searchEngineElasticsearch) { %>Search<% } %>Repository.search(query)<%_ if (dtoMapstruct) { %>
-            .map(<%= entityToDtoReference %>)<%_ } %><%_ if (fromResource) { _%>.collectList()<%_ } _%>;
+            return <%= entityInstance %><% if (searchEngineElasticsearch) { %>Search<% } %>Repository.search(query)<%_ if (dtoMapstruct) { %>
+                .map(<%= entityToDtoReference %>)<%_ } %><%_ if (fromResource) { _%>.collectList()<%_ } _%>;
   <%_ } else { _%>
     <%_ if (searchEngineElasticsearch) { _%>
-        return StreamSupport
-            .stream(<%= entityInstance %>SearchRepository.search(query).spliterator(), false)
+            return StreamSupport
+                .stream(<%= entityInstance %>SearchRepository.search(query).spliterator(), false)
     <%_ } else { _%>
-            return <%= entityInstance %>Repository.search(query)<% if (!dtoMapstruct) { %>;<% } else { %>.stream()<% } %>
+                return <%= entityInstance %>Repository.search(query)<% if (!dtoMapstruct) { %>;<% } else { %>.stream()<% } %>
     <%_ } _%>
     <%_ if (dtoMapstruct) { _%>
-            .map(<%= entityToDtoReference %>)
+                .map(<%= entityToDtoReference %>)
     <%_ } _%>
     <%_ if (dtoMapstruct || searchEngineElasticsearch) { _%>
-            .toList();
+                .toList();
     <%_ } _%>
   <%_ } _%>
 <%_ } else { _%>
-        return <%= entityInstance %>Service.search(query)<% if (reactive && fromResource) { %>.collectList()<% } %>;
+            return <%= entityInstance %>Service.search(query)<% if (reactive && fromResource) { %>.collectList()<% } %>;
 <%_ } _%>
+        } catch (RuntimeException e) {
+    <% if (searchEngineElasticsearch && fromResource) { %>
+            throw ElasticsearchExceptionMapper.mapException(e);
+    <%_ } else { _%>
+            throw e;
+    <% } %>
+        }

--- a/generators/server/templates/entity/src/main/java/package/common/search_template.ejs
+++ b/generators/server/templates/entity/src/main/java/package/common/search_template.ejs
@@ -37,13 +37,21 @@ if (paginationNo) { %>
             .map(page -> PaginationUtil.generatePaginationHttpHeaders(UriComponentsBuilder.fromHttpRequest(request), page))
             .map(headers -> ResponseEntity.ok().headers(headers).body(<%= entityInstance %><%= viaService ? 'Service' : (searchEngineElasticsearch ? 'Search' : '') + 'Repository' %>.search(query, pageable)<% if (!viaService && dtoMapstruct) { %>.map(<%= entityToDtoReference %>)<% } %>));
   <%_ } else { _%>
+        try {
     <%_ if (viaService) { _%>
-        Page<<%= instanceType %>> page = <%= entityInstance %>Service.search(query, pageable);
+            Page<<%= instanceType %>> page = <%= entityInstance %>Service.search(query, pageable);
     <%_ } else { _%>
-        Page<<%= persistClass %>> page = <%= entityInstance %><% if (searchEngineElasticsearch) { %>Search<% } %>Repository.search(query, pageable);
+            Page<<%= persistClass %>> page = <%= entityInstance %><% if (searchEngineElasticsearch) { %>Search<% } %>Repository.search(query, pageable);
     <%_ } _%>
-        HttpHeaders headers = PaginationUtil.generatePaginationHttpHeaders(<% if (reactive) { %>UriComponentsBuilder.fromHttpRequest(request)<% } else { %>ServletUriComponentsBuilder.fromCurrentRequest()<% } %>, page);
-        return ResponseEntity.ok().headers(headers).body(<% if (!viaService && dtoMapstruct) { %><%= entityListToDtoListReference %>(<% } %>page.getContent()<% if (!viaService && dtoMapstruct) { %>)<% } %>);
+            HttpHeaders headers = PaginationUtil.generatePaginationHttpHeaders(<% if (reactive) { %>UriComponentsBuilder.fromHttpRequest(request)<% } else { %>ServletUriComponentsBuilder.fromCurrentRequest()<% } %>, page);
+            return ResponseEntity.ok().headers(headers).body(<% if (!viaService && dtoMapstruct) { %><%= entityListToDtoListReference %>(<% } %>page.getContent()<% if (!viaService && dtoMapstruct) { %>)<% } %>);
+        } catch (RuntimeException e) {
+    <% if (searchEngineElasticsearch) { %>
+            throw ElasticsearchExceptionMapper.mapException(e);
+    <%_ } else { _%>
+            throw new RuntimeException(e);
+    <% } %>
+        }
   <%_ } _%>
 <% } -%>
     }

--- a/generators/server/templates/entity/src/main/java/package/web/rest/_EntityClass_Resource.java.ejs
+++ b/generators/server/templates/entity/src/main/java/package/web/rest/_EntityClass_Resource.java.ejs
@@ -39,6 +39,9 @@ import <%= entityAbsolutePackage %>.repository.search.<%= entityClass %>SearchRe
 import <%= entityAbsolutePackage %>.repository.UserRepository;
 <%_ } _%>
 import <%= packageName %>.web.rest.errors.BadRequestAlertException;
+<%_ if (searchEngineElasticsearch) { _%>
+import <%= packageName %>.web.rest.errors.ElasticsearchExceptionMapper;
+<%_ } _%>
 <%_ if (dtoMapstruct) { _%>
 import <%= entityAbsolutePackage %>.service.dto.<%= dtoClass %>;
   <%_ if (serviceNo) { _%>

